lib/ansible/plugins/lookup/gcp_lookup.py

```diff
@@ -0,0 +1,327 @@
+  - requests >= 2.18.4
+  - google-auth >= 1.3.0
+extends_documentation_fragment:
+  - gcp
+short_description: Look up and verify GCP Attributes
+description:
+  - Describes attributes on GCP. You can specify one of the listed
+    attribute choices or omit it to see all attributes.
+options:
+  attribute:
+    description: The attribute for which to get the value(s).
+    default: null
+    choices:
+      - disk_type
+      - license
+      - machine_type
+      - region
+      - zone
+  return:
+    description: An optional value to describe what part of the attribute should
+                 be returned
+    default: null
+"""
+
+EXAMPLES = """
+vars:
+  account_details: "{{ lookup('gcp_lookup',
+                              attribute='region',
+                              label='us-west1',
+                              auth_kind='serviceaccount',
+                              service_account_file='/tmp/my_account.json',
+                              project='test-project',
+                              scopes=['https://www.googleapis.com/auth/compute']) }}"
+  # us-west1
+  account_details: "{{ lookup('gcp_lookup',
+                              attribute='region',
+                              return='self_link',
+                              label='us-west1',
+                              auth_kind='serviceaccount',
+                              service_account_file='/tmp/my_account.json',
+                              project='test-project',
+                              scopes=['https://www.googleapis.com/auth/compute']) }}"
+  # us-west1
+
+"""
+
+RETURN = """
+"""
+
+
+################################################################################
+# Imports
+################################################################################
+
+from ansible.errors import AnsibleError
+from ansible.plugins import AnsiblePlugin
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils.gcp_utils import navigate_hash, GcpSession, GcpRequestException
+import json
+
+################################################################################
+# Main
+################################################################################
+
+
+class GcpModule(object):
+    def __init__(self, options):
+        self.params = options
+        if 'label' in self.params:
+            self.params['name'] = self.params['label']
+            del self.params['label']
+
+    def fail_json(self, **kwargs):
+        raise AnsibleError(kwargs['msg'])
+
+
+class GcpDiskType(object):
+    def __init__(self, options):
+        self.module = GcpModule(options)
+
+        self.kind = 'compute#diskType'
+        self.link = "https://www.googleapis.com/compute/v1/projects/{project}/zones/{zone}/diskTypes/{name}".format(**self.module.params)
+
+    def _fetch_resource(self):
+        auth = GcpSession(self.module, 'compute')
+        return self._return_if_object(auth.get(self.link))
+
+    def _return_if_object(self, response):
```

Comment:
This looks great! I noticed it looks like this method is identically duplicated 5 times. You could make it common code instead.